### PR TITLE
use proper docker protobufs and fix rlimits NPE

### PR DESCRIPTION
**Pull request by mgummelt**
cc @mrbrowning

### CODE REVIEW

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/RawContainer.java [hunk 1]
```diff
@@ -15,7 +16,12 @@ private RawContainer(
             @JsonProperty("image-name") String imageName,
             @JsonProperty("rlimits") WriteOnceLinkedHashMap<String, RawRLimit> rlimits) {
         this.imageName = imageName;
-        this.rlimits = rlimits;
+
+        if (rlimits != null) {
```
Comment (mgummelt) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/RawContainer.java]: There may be a way of more gracefully handling this through the YAML mapper, but this was quicker.

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/RawTask.java [hunk 2]
```diff
@@ -76,10 +73,6 @@ public String getCmd() {
         return cmd;
     }
 
-    public String getImage() {
```
Comment (mgummelt) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/RawTask.java]: was never used, and I don't believe it makes sense to specify an image on a task

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/RawContainer.java (hunk 1 above)
Comment (mrbrowning) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/RawContainer.java]: From what I've seen, the typical pattern for dealing with potentially `null` members in the raw YAML specs has been to check for emptiness at the site of access, e.g. [YAMLToInternalMappers.java#L248](https://github.com/mesosphere/dcos-commons/blob/13165ea10fe67f4cc49ffaf09f95617eebe7a2c7/sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/YAMLToInternalMappers.java#L248). I'm not married to it, but it'd be good to have the approach be consistent.

Comment (mgummelt) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/RawContainer.java]: Those checks are for emptiness, not non-nullness, right?  All else equal, I think it's better to get rid of nulls early.

Comment (nickbp) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/RawContainer.java]: The util functions being used in there also check for null FYI

(I'm accustomed to `isEmptyOrNull()` but I can see why they'd opt for more brevity)

Comment (mgummelt) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/RawContainer.java]: Yea, I know.  My point is that those fields aren't null.  They're empty collections.  That's what I'm trying to do here as well.

Comment (mgummelt) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/RawContainer.java]: Jackson will auto-translate missing fields to empty collections if they're a `Collection`, but a `HashMap` is not a collection, which is why we need to set it here explicitly for consistency.

Comment (mrbrowning) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/RawContainer.java]: Makes sense. I agree about reducing the scope of a null field, just wanted to ensure consistency in one direction or another.

**Pull request merged**